services/ui/src/components/Honeycomb/index.js

```diff
@@ -0,0 +1,189 @@
+import React, {useState, Fragment, useEffect, useRef} from "react";
+import { HexGrid, Layout, Hexagon, Text, GridGenerator, HexUtils } from 'react-hexgrid';
+import * as R from 'ramda';
+import ProblemsByProject from "components/ProblemsByProject";
+import {LoadingPageNoHeader} from 'pages/_loading';
+import {ErrorNoHeader} from 'pages/_error';
+import { bp } from 'lib/variables';
+import './styling.css';
+
+const config = {
+    "width": 1200,
+    "height": 100,
+    "layout": {"width": 4, "height": 4, "flat": false, "spacing": 1.08},
+    "origin": {"x": 0, "y": 0},
+    "map": "rectangle",
+};
+
+const Honeycomb = ({ data, filter }) => {
+    const { projectsProblems } = data || [];
+    const [projects, setProjects] = useState(projects);
+    const [projectInView, setProjectInView] = useState(false);
+    const [display, setDisplay] = useState({type: "normal", multiplier: 2});
+
+    const generator = GridGenerator.getGenerator(config.map);
+    const projectCount = projectsProblems && parseInt(projectsProblems.length);
+    const displayMultiple = display && parseInt(display.multiplier * 8);
+    let rows = projectsProblems && parseInt(projectCount / displayMultiple);
+
+    const hexs = generator.apply(config, [displayMultiple, ++rows]);
+    const layout = config.layout;
+    const size = {
+        x: parseInt(display.hexSize * layout.width),
+        y: parseInt(display.hexSize * layout.height)
+    };
+
+    const handleHexClick = (project) => {
```

Comment:
This here seems to suggest that this Honeycomb component is pretty closely tied with the project and environments stuff - which suggests to me we should better compartmentalize (as per my earlier comment)

Reply:
👍 yep - moved this to inside problemsByProject component as it is only going to be used there.